src/lualib/ArrayUnshift.ts

```diff
@@ -1,6 +1,12 @@
-export function __TS__ArrayUnshift<T>(this: void, arr: T[], ...items: T[]): number {
-    for (let i = items.length - 1; i >= 0; --i) {
-        table.insert(arr, 1, items[i]);
+export function __TS__ArrayUnshift<T>(this: T[], ...items: T[]): number {
+    const length = items.length;
```

Comment:
Length is confusing here because it's not `this.length`, suggest  to rename to `numItemsToInsert`